spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/annotation/web/configurers/AuthorizationServerSecurityConfigurer.java

```diff
@@ -208,10 +208,7 @@ private ClientCredentialsTokenEndpointFilter clientCredentialsTokenEndpointFilte
 				frameworkEndpointHandlerMapping().getServletPath("/oauth/token"));
 		clientCredentialsTokenEndpointFilter
 				.setAuthenticationManager(http.getSharedObject(AuthenticationManager.class));
-		OAuth2AuthenticationEntryPoint authenticationEntryPoint = new OAuth2AuthenticationEntryPoint();
-		authenticationEntryPoint.setTypeName("Form");
-		authenticationEntryPoint.setRealmName(realm);
-		clientCredentialsTokenEndpointFilter.setAuthenticationEntryPoint(authenticationEntryPoint);
+		clientCredentialsTokenEndpointFilter.setAuthenticationEntryPoint(this.authenticationEntryPoint);
```

Comment:
@g-u-a-n  It doesn't look like the logic here is correct. 

The `configure()` method has this conditional check:

```java
	if (allowFormAuthenticationForClients) {
		clientCredentialsTokenEndpointFilter(http);
	}
```

If `POST` authentication is allowed for clients then configure `ClientCredentialsTokenEndpointFilter` with `this.authenticationEntryPoint`. The custom `AuthenticationEntryPoint` can be supplied via `authenticationEntryPoint()` and is ultimately configured (`init()`) as the `AuthenticationEntryPoint` for ALL protected endpoints (`HttpSecurity`) of the Authorization Server  NOT just `/oauth/token`.

The `AuthenticationEntryPoint` configured in `clientCredentialsTokenEndpointFilter()` cannot be the same `AuthenticationEntryPoint` used in `init()` -> `http.httpBasic().authenticationEntryPoint()`.

I also see there are no tests that go with this change. This is required with any proposed changes. If you would like to re-work this and add tests we can consider merging it.

I'm going to close this but feel free to submit a new PR.

As a workaround, you can explicitly configure the `ClientCredentialsTokenEndpointFilter` and directly add it via `httpSecurity.addFilterBefore(clientCredentialsTokenEndpointFilter, BasicAuthenticationFilter.class)`